tee-worker/litentry/core/vc-issuance/lc-vc-task-receiver/src/lib.rs

```
@@ -208,19 +231,19 @@ where
 			response.vc_hash,
 			H256::zero(),
 		));
-		let res = RequestVCResult {
-			vc_index: response.vc_index,
-			vc_hash: response.vc_hash,
-			vc_payload: result,
-		};
+
 		// This internally fetches nonce from a Mutex and then updates it thereby ensuring ordering
-		let xt = extrinsic_factory
-			.create_extrinsics(&[call], None)
-			.map_err(|e| format!("Failed to construct extrinsic for parentchain: {:?}", e))?;
-		context
-			.ocall_api
-			.send_to_parentchain(xt, &ParentchainId::Litentry, false)
-			.map_err(|e| format!("Unable to send extrinsic to parentchain: {:?}", e))?;
+		// Only log the error message even if it fails as `res` is already available to the client
```

We are not awaiting for the extrinsic to be included in a block, that is why the `false` in `send_to_parentchain(xt, &ParentchainId::Litentry, false)`.I think this begs the question of how important is sending an extrinsic to the parachain  and what are the consequence of not sending an extrinsic to parachain, and if none, I'm wondering why is there even a need to submit an `extrinsic`

---

The conclusion was drawn from [this doc](https://www.notion.so/web3builders/Collection-of-open-discussions-2023-08-2299815065a24b3fbc9043194e5738cd#c5c395f513c64818b79ae536e1d096fa).

Basically we use these events for tracking and statistic purpose. Concretely, we can index the `vc_issued` events to know e.g. the requester count distribution for each assertion type. So it's basically acceptable that we are (somehow) missing these events randomly, but if there's a large amount of event failure, we should look into it.

`send_to_parentchain` can fail early if e.g. we have the wrong call_index even with `false` in parameter, I tried to be more lenient with such cases

---

So every time we create an extrinsic(in vc-issuance) we utilize a Mutex to get the nonce(of worker), and this slows down the response time because every request essentially waits for this mutex to get unlocked. I was just wondering if it is inconsequential, it would be good to move away from it. (or Alternatively we don't have to send one extrinsic at a time and do batch). This might be out of scope for the PR, but just want to hear your thoughts on this.

---

Yes I agree it would be good to remove it or do it aysnc

---

one sec - I think nonce mutex acquiring should be quite light-weighted, and I don't see any time-consuming job during the lock holding period, so it _should_ be fine most of the time. It's still better to optimise it away though 🤔

---

Created: P-393